framework/ui/view/launchprovider.cpp

```diff
@@ -137,7 +163,66 @@ RetVal<Val> QmlLaunchProvider::toRetVal(const QVariant& jsrv) const
     return rv;
 }
 
-void QmlLaunchProvider::onOpen(QString pageType)
+RetVal<LaunchProvider::OpenData> LaunchProvider::openWidget(const UriQuery &q)
```

Comment:
Better rename `openWidget` to `openWidgetDialog`
because only QDialog possible to open